File: calculations/python/currency_conversion.py
# Pratice 11. Currency Conversion
# Output:
#   How many euros are you exchanging? 81
#   What is the exchange rate? 134.81
#   81 euros at an exchange rate of 134.81
#   109.20 U.S. dollars.
# Constraint:
#   - Ensure that fractions of a cent are rounded up to the next penny.
#   - Use a single output statement.

#!/usr/bin/env python

import sys

def input_process(in_question):
    return input(in_question) if sys.version_info >= (3,0) else raw_input(in_question)


if __name__ == '__main__':
    try:
        euros = int(input_process('How many euros are you exchanging? '))
        exchange_rate = float(input_process('What is the exchange rate? '))
    except:
        print ('You must input only numbers.')
    else:
        if euros <= 0:
            print ('You must enter a positive number.')
            exit(1)

        dollars = round((euros * exchange_rate / 100), 2)

        print_str = '''
        %s euros at an exchange rate of %s
        %s U.S. dollars.
        ''' % (euros, exchange_rate, dollars)

        print (print_str)

File: calculations/go/currency_conversion.go
/*
 * Pratice 11. Currency Conversion
 * Output:
 *   How many euros are you exchanging? 81
 *   What is the exchange rate? 134.81
 *   81 euros at an exchange rate of 134.81
 *   109.20 U.S. dollars.
 * Constraint:
 *   - Ensure that fractions of a cent are rounded up to the next penny.
 *   - Use a single output statement.
 */

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("How many euros are you exchanging?  ")
	euros_str, _ := reader.ReadString('\n')
	fmt.Println("What is the exchange rate? ")
	exchange_rate_str, _ := reader.ReadString('\n')

	euros, _ := strconv.ParseFloat(strings.TrimSpace(euros_str), 64)
	exchange_rate, _ := strconv.ParseFloat(strings.TrimSpace(exchange_rate_str), 64)

	if euros <= 0 {
		fmt.Println("You must enter a positive number.")
		os.Exit(1)
	}

	dollars := euros * exchange_rate / 100

	print_str := fmt.Sprintf("%.2f euros at an exchange rate of %.2f\n", euros, exchange_rate)
	print_str += fmt.Sprintf("%.2f U.S. dollars.", dollars)

	fmt.Println(print_str)
}
